packages/workflow/src/internals.ts

```diff
@@ -638,7 +664,7 @@ export class Activator implements ActivationHandler {
     const doUpdateImpl = async () => {
       let input: UpdateInput;
       try {
-        if (runValidator && this.updateHandlers.get(name)?.validator) {
+        if (runValidator && entry.validator) {
```

Comment:
Just below:

```
this.validateUpdateNextHandler.bind(this, entry.validator)
```

And add the corresponding argument on `validateUpdateNextHandler`. Then remove the access on `this.updateHandlers` from `validateUpdateNextHandler`.

Reply:
Beautiful. I don't have a ton of JS background and I always forget that bind can be used for general partial application like this. I really like this improvement. 77240fe49fddf2fc8c5b91cd9a2c2daf9d34550f